include/zephyr/location.h

```diff
@@ -0,0 +1,402 @@
+/**
+ * @file location.h
+ *
+ * @brief Public location API.
+ *
+ */
+
+/*
+ * Copyright (c) 2022, Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LOCATION_H_
+#define ZEPHYR_INCLUDE_LOCATION_H_
+
+/**
+ * @brief Location Interface
+ * @defgroup location_interface Location Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Location data
+ *
+ * The latitude and longitude are in the unit micro degree. 1 micro
+ * degree corresponds to approx. 11cm along the earths surface.
+ *
+ * The accuracy describes the radius within which the real
+ * position is with a 68.2% certainty, in millimeters.
+ */
+struct location_position {
+	/** Latitudal position in microdegrees (0 to +-180000000) */
```

Comment:
While microdegrees is a nice unit, I know that UBLOX GNSS modems at least return degrees with `10^-7` scaling.
11cm is probably fine for GNSS, but more local methods of determining location can do much better than that.
As this is a generic location API, I doubt 11cm is sufficient.

Reply:
@RobMeades

Reply:
We can have it nano degrees as well, then we will have to bump it up to 64 bit however :)

Reply:
Changed units to nanodegrees, stored in int64_t